### DISCUSSION TITLE
SDC Solution for Business Demography using Tau-Argus

### DISCUSSION THREAD

**Discussion by KarinaKelleher** [sdcTools/UserSupport#214] [User Group]
We (Tim Linehan and Karina Dineen) in Methodology in CSO in Ireland are currently finalizing an SDC solution to apply suppression to Business Demography deliverables 9A to 9H and 9M using Tau-Argus. Due to the presence of non-nested hierarchies (S94 etc) we have-to run Tau-Argus 3 times for each variable using the different nested definitions (created from the non-nested definitions). 

However, also due to the Eurostat requirement to specify which dominance fails are (1,80) fails and which are (2,90) fails and, as Tau-Argus doesn't differentiate between these, we are now running Tau-Argus  first of all to get (1,80) fails only, then next to get (2,90) fails only and finally to check for all 3 Primary rules (1,80), (2,90) and min. freq. of 3 and then combining the relevant details. This in turn means 3 runs for each hierarchical definition for each variable - a total of 9 runs for each variable.

Basically we are wondering if Tau-Argus can in fact differentiate between which dominance rule fail is causing a Primary Suppression or if others have found a less cumbersome solution for this requirement?
Thanks.

**Comment by ppdewolf** [comment]
Concerning your second paragraph: this sounds like the SBS format that Eurostat requires/required. In Tau-Argus there is a possibility to save the table in "sbs"-format. It is not the complete/exact SBS-format that Eurostat requests, but has at least the following features that may be interesting to you:
it adds the following status to each cell:
A = frequency unsafe
B = Dominance unsafe with one contributor
C = Dominance unsafe with two contributors
D = Secondary unsafe
V = Safe
In case of B and C it will also includes the percentage of the contributor(s) that dominate.
See page 102-103 in the manual of Tau-Argus.

**Comment by KarinaKelleher** [comment]
Thank you for your reply.

We have run it using this method and yes definitely this feature is what we required. It now means that we will only need to run through each hierarchy once (instead of 3) times so has cut the workload to one-third! This will make a big difference as the section will be dealing with almost 100 variables.

Thanking you again,
Karina and Tim.

**Comment by KarinaKelleher** [comment]
While using the SBS format we noticed that missing values are not written to the output table even when the 'Suppress empty cells' box in unchecked in General Options on the 'Save table' screen. We are using version 4.2.0 Build 5.
Do we need to do something differently to output these?

**Comment by ppdewolf** [comment]
@KarinaKelleher: You are correct that empty cells are not written to the output file when using the SBS format. I just checked the code and it is hard coded that way. It just means that whenever a cell is not in the output, it is thus empty.

I am not really familiar with the SBS format, so perhaps someone else who is more familiar with that format can tell you whether this is actually part of the format as issued by Eurostat (leaving empty cells out of the table file send to Eurostat).

A work around may be that you save the output twice and combine the output: as SBS format to get the flags for dominance by 1, 2 or threshold, and as e.g. Code-value with "add status". the status 14 is then "empty".
You only need to run the protection once. After that you can save the output in as many different formats as you like.

**Comment by KarinaKelleher** [comment]
@ppdewolf Thank you for your reply - That work around should be sufficient for our needs - And may not even be necessary depending on the requirements of the area. We were tracking the number of rows just to ensure that no valid information had been deleted when filtering on the safe file.

**Comment by violetacln** [comment]
Hi, Karina @KarinaKelleher ! I was sending the signed up participants a message with details/link for the coming User Meeeting and the message for you bounced back, it seems the email I have from you is not working? Please, let me know what is the correct email :)

**Comment by tdlinehancso** [comment]
Violetta,

Tim Linehan here - I work with Karina. What email address did you use for her? Have included her email address in the CC

By the way, I didn't get that zoom link for the meeting either?

Tim

Tim Linehan,
Methodology Division,
Central Statistics Office

"Violeta Calian" ---21/02/2022 14:20:17---CAUTION: This email originated from outside of the CSO. Do not click links or open   attachments unl

From:	"Violeta Calian" ***@***.***>
To:	
Cc:	"Subscribed" ***@***.***>
Date:	21/02/2022 14:20
Subject:	Re: [sdcTools/user-group] SDC Solution for Business Demography using Tau-Argus (#9)





CAUTION: This email originated from outside of the CSO. Do not click links or open attachments unless you recognise the sender and know the content is safe. If in doubt contact the Service Desk at Extension 5666.





Hi, Karina! I was sending the signed up participants a message with details/link for the coming User Meeeting and the message for you bounced back, it seems the email I have from you is not working? Please, let me know what is the correct email :)
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by KarinaKelleher** [comment]
Good morning Violeta,

I have received the relevant agenda and meeting link - I'm not sure why you received the email bounce back,

Regards,
Karina.

Karina Dineen
Statistician
Methodology

------------------------------------------------
 
Central Statistics Office | Skehard Road, | Cork, | T12 XOOE | Ireland. 
Tel +353 21 453 5716   

www.cso.ie




The CSO are currently conducting a user satisfaction survey. 
Insights you provide will help us enhance and improve our services to the public.
Have your say. Take part in our short survey → https://bit.ly/CSOsatisfactionsurvey



"Violeta Calian" ---21/02/2022 14:20:15---CAUTION: This email originated from outside of the CSO. Do not click links or open   attachments unl

From:	"Violeta Calian" ***@***.***>
To:	
Cc:	"KarinaKelleher" ***@***.***>, "Mention" ***@***.***>
Date:	21/02/2022 14:20
Subject:	Re: [sdcTools/user-group] SDC Solution for Business Demography using Tau-Argus (#9)





CAUTION: This email originated from outside of the CSO. Do not click links or open attachments unless you recognise the sender and know the content is safe. If in doubt contact the Service Desk at Extension 5666.





Hi, Karina! I was sending the signed up participants a message with details/link for the coming User Meeeting and the message for you bounced back, it seems the email I have from you is not working? Please, let me know what is the correct email :)
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by KarinaKelleher** [comment]
Hi Violeta,

This is my email address – ***@***.******@***.***>
Perhaps our email system just rejected the email address it was coming from –if it happens again you could let me know what email address this would be from and I can ask them to stop blocking emails from that address,

Regards,
Karina.

**Comment by KarinaKelleher** [comment]
Karina Dineen would like to recall the message, "[sdcTools/UserSupport] SDC Solution for Business Demography using Tau-Argus (Discussion #214)".
https://www.cso.ie/en/emaildisclaimer/

**Comment by KarinaKelleher** [comment]
Hi Violeta,

This is my email address – ***@***.******@***.***>
I think what may be happening is that our email system is rejecting the email address it was coming from –if it happens again could you could let me know what email address this would be from and I can ask them to stop blocking emails from that address,

Regards,
Karina.
